File: backend/app/api/v1/professionals.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.dependencies import get_current_user, get_db
from app.models.user import User
from app.schemas.common import PaginatedResponse
from app.schemas.professional import (
    ProfessionalListItem,
    ProfessionalProfileResponse,
    ProfessionalProfileUpdate,
)
from app.services.professional_profile_service import ProfessionalProfileService

router = APIRouter(prefix="/professionals", tags=["professionals"])


@router.get("", response_model=PaginatedResponse[ProfessionalListItem])
async def list_professionals(
    page: int = Query(1, ge=1),
    per_page: int = Query(20, ge=1, le=100),
    especialidad: str | None = Query(None),
    q: str | None = Query(None),
    sort_by: str | None = Query(None),
    sort_order: str = Query("desc"),
    db: AsyncSession = Depends(get_db),
):
    service = ProfessionalProfileService(db)
    items, total = await service.list_professionals(page, per_page, especialidad, q, sort_by, sort_order)
    total_pages = max(1, (total + per_page - 1) // per_page)
    return PaginatedResponse(
        items=items,
        total=total,
        page=page,
        per_page=per_page,
        total_pages=total_pages,
    )


@router.get("/specialties")
async def list_specialties(db: AsyncSession = Depends(get_db)):
    specialties = await ProfessionalProfileService(db).list_specialties()
    return {"items": specialties}


@router.get("/me", response_model=ProfessionalProfileResponse)
async def get_my_profile(
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    return await ProfessionalProfileService(db).get_my_profile(current_user.id)


@router.put("/me", response_model=ProfessionalProfileResponse)
async def update_my_profile(
    data: ProfessionalProfileUpdate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    return await ProfessionalProfileService(db).update_my_profile(current_user.id, data)

File: frontend/src/api/professionals.ts
import client from './client';
import type { PaginatedResponse, ProfessionalListItem, ProfessionalProfile } from '@/types';

export async function listProfessionals(
  page = 1,
  perPage = 20,
  especialidad?: string,
  q?: string,
  sortBy?: string,
  sortOrder?: string,
): Promise<PaginatedResponse<ProfessionalListItem>> {
  const params = new URLSearchParams({ page: String(page), per_page: String(perPage) });
  if (especialidad) params.set('especialidad', especialidad);
  if (q) params.set('q', q);
  if (sortBy) params.set('sort_by', sortBy);
  if (sortOrder) params.set('sort_order', sortOrder);
  const res = await client.get<PaginatedResponse<ProfessionalListItem>>(
    `/professionals?${params.toString()}`,
  );
  return res.data;
}

export async function listSpecialties(): Promise<{ items: string[] }> {
  const res = await client.get<{ items: string[] }>('/professionals/specialties');
  return res.data;
}

export async function getMyProfessionalProfile(): Promise<ProfessionalProfile> {
  const res = await client.get<ProfessionalProfile>('/professionals/me');
  return res.data;
}

export async function updateMyProfessionalProfile(
  data: Partial<ProfessionalProfile>,
): Promise<ProfessionalProfile> {
  const res = await client.put<ProfessionalProfile>('/professionals/me', data);
  return res.data;
}
